src/release/mod.rs

```diff
@@ -0,0 +1,472 @@
+/*!
+PGXN release `META.json` validation and management.
+
+This module provides interfaces to load, validate, and manipulate PGXN release
+`META.json` files. PGXN adds release metadata to distribution-provided
+[v1] and [v2] `META.json` data to identify the user who made a release, the
+timestamp, hash digests of the release file. In [v2], it also includes a
+download URI and a private key signature.
+
+Use [`Distribution`] to validate the `META.json` file included in a
+distribution.
+
+It supports both the [v1] and [v2] specs.
+
+  [v1]: https://rfcs.pgxn.org/0001-meta-spec-v1.html
+  [v2]: https://github.com/pgxn/rfcs/pull/3
+
+*/
+
+use crate::meta::*;
+use crate::util;
+
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::{borrow::Borrow, collections::HashMap, error::Error, fs::File, path::Path};
+
+/// Digests represents Hash digests for a file that can be used to verify its
+/// integrity.
+#[serde_with::serde_as]
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
+pub struct Digests {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde_as(as = "Option<serde_with::hex::Hex>")]
+    sha1: Option<[u8; 20]>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde_as(as = "Option<serde_with::hex::Hex>")]
+    sha256: Option<[u8; 32]>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde_as(as = "Option<serde_with::hex::Hex>")]
+    sha512: Option<[u8; 64]>,
+}
+
+impl Digests {
+    /// Borrows the SHA-1 hash.
+    pub fn sha1(&self) -> Option<&[u8]> {
```

Comment:
Another option:

```suggestion
    pub fn sha1(&self) -> Option<&[u8; 20]> {
        self.sha1.as_ref()
    }
```

Reply:
Ahhh, thank you! Switched to that in 88b2183.